Firestore/core/test/firebase/firestore/local/lru_garbage_collector_test.cc

```diff
@@ -0,0 +1,740 @@
+  // later, which we track here.
+  DocumentKeySet middle_docs_to_remove;
+  // This will be the document in this target that gets an update later
+  DocumentKey middle_doc_to_update;
+  QueryData middle_target =
+      persistence_->Run("Add middle target and docs", [&] {
+        QueryData middle_target = AddNextQueryInTransaction();
+        // these docs will be removed from this target later, triggering a bump
+        // to their sequence numbers. Since they will not be a part of the
+        // target, we expect them to be removed.
+        for (int i = 0; i < 2; i++) {
+          Document doc = CacheADocumentInTransaction();
+          expected_removed.insert(doc.key());
+          AddDocument(doc.key(), middle_target.target_id());
+          middle_docs_to_remove = middle_docs_to_remove.insert(doc.key());
+        }
+        // these docs stay in this target and only this target. There presence
+        // in this target prevents them from being GC'd, so they are also
+        // expected to be retained.
+        for (int i = 2; i < 4; i++) {
+          Document doc = CacheADocumentInTransaction();
+          expected_retained.insert(doc.key());
+          AddDocument(doc.key(), middle_target.target_id());
+        }
+        // This doc stays in this target, but gets updated.
+        {
+          Document doc = CacheADocumentInTransaction();
+          expected_retained.insert(doc.key());
+          AddDocument(doc.key(), middle_target.target_id());
+          middle_doc_to_update = doc.key();
+        }
+        return middle_target;
+      });
+
+  // Add the newest target and add 5 documents to it. Some of those documents
+  // will additionally be added to the oldest target, which will cause those
+  // documents to be retained. The remaining documents are expected to be
+  // removed, since this target will be removed.
+  DocumentKeySet newest_docs_to_add_to_oldest;
+  persistence_->Run("Add newest target and docs", [&] {
+    QueryData newest_target = AddNextQueryInTransaction();
+    // These documents are only in this target. They are expected to be removed
+    // because this target will also be removed.
+    for (int i = 0; i < 3; i++) {
+      Document doc = CacheADocumentInTransaction();
+      expected_removed.insert(doc.key());
+      AddDocument(doc.key(), newest_target.target_id());
+    }
+    // docs to add to the oldest target in addition to this target. They will be
+    // retained
+    for (int i = 3; i < 5; i++) {
+      Document doc = CacheADocumentInTransaction();
+      expected_retained.insert(doc.key());
+      AddDocument(doc.key(), newest_target.target_id());
+      newest_docs_to_add_to_oldest =
+          newest_docs_to_add_to_oldest.insert(doc.key());
+    }
+  });
+
+  // 2 doc writes, add one of them to the oldest target.
+  persistence_->Run("2 doc writes, add one of them to the oldest target", [&] {
+    // write two docs and have them ack'd by the server. can skip mutation queue
```

Comment:
Ultranit: capitalize the first two sentences.

Reply:
Done.